airflow/providers/amazon/aws/transfers/s3_to_sql.py

```diff
@@ -0,0 +1,158 @@
+        If `column_list` is set to 'infer', this is ignored
+    :param commit_every: The maximum number of rows to insert in one
+        transaction. Set to `0` to insert all rows in one transaction.
+    :param csv_reader_kwargs: key word arguments to pass to csv.reader().
+        This lets you control how the CSV is read.
+        e.g. To use a different delimiter, pass the following dict:
+        {'delimiter' : ';'}
+    """
+
+    template_fields: Sequence[str] = (
+        "s3_bucket",
+        "s3_key",
+        "schema",
+        "table",
+        "column_list",
+        "sql_conn_id",
+    )
+    template_ext: Sequence[str] = ()
+    ui_color = "#f4a460"
+
+    def __init__(
+        self,
+        *,
+        s3_key: str,
+        s3_bucket: str,
+        table: str,
+        column_list: Literal["infer"] | list[str] | None = "infer",
+        commit_every: int = 1000,
+        schema: str | None = None,
+        skip_first_row: bool = False,
+        sql_conn_id: str = "sql_default",
+        aws_conn_id: str = "aws_default",
+        csv_reader_kwargs: dict[str, Any] | None = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.s3_bucket = s3_bucket
+        self.s3_key = s3_key
+        self.table = table
+        self.schema = schema
+        self.aws_conn_id = aws_conn_id
+        self.sql_conn_id = sql_conn_id
+        self.column_list = column_list
+        self.commit_every = commit_every
+        self.skip_first_row = skip_first_row
+        if csv_reader_kwargs:
+            self.csv_reader_kwargs = csv_reader_kwargs
+        else:
+            self.csv_reader_kwargs = {}
+
+    def execute(self, context: Context) -> None:
+
+        self.log.info("Loading %s to SQL table %s...", self.s3_key, self.table)
+
+        s3_hook = S3Hook(aws_conn_id=self.aws_conn_id)
+        self._file = s3_hook.download_file(key=self.s3_key, bucket_name=self.s3_bucket)
+
+        hook = self._get_hook()
+        try:
+            # open with newline='' as recommended
+            # https://docs.python.org/3/library/csv.html#csv.reader
+            with open(self._file, newline="") as file:
+                reader = csv.reader(file, **self.csv_reader_kwargs)
+
+                if self.column_list == "infer":
+                    self.column_list = list(next(reader))
+                    self.log.info("Column Names inferred from csv: %s", self.column_list)
+                elif self.skip_first_row:
+                    next(reader)
+
+                hook.insert_rows(
+                    table=self.table,
+                    schema=self.schema,
+                    target_fields=self.column_list,
+                    rows=reader,
+                    commit_every=self.commit_every,
+                )
+
+        finally:
+            # Remove file downloaded from s3 to be idempotent.
+            os.remove(self._file)
+
+    def _get_hook(self) -> DbApiHook:
```

Comment:
nit. You can decorate this function with @cached_property

Reply:
Perhaps this doesn't need to be a property at all since it's only used in the `execute()` method?